numpy/__init__.pyi

```diff
@@ -2454,14 +2364,9 @@ class ndarray(_ArrayOrScalarCommon, Generic[_ShapeType, _DType_co]):
     def __pow__(self: NDArray[object_], other: Any) -> Any: ...
     @overload
     def __pow__(self: NDArray[Any], other: _ArrayLikeObject_co) -> Any: ...
-    @overload
-    def __pow__(
-        self: NDArray[Union[bool_, number[Any]]],
-        other: _RecursiveSequence,
-    ) -> Any: ...
```

Comment:
Don't need a catchall overload for >4D nested sequences anymore as we can now represent arbitrary levels of nesting.